qdb/tracer.py

```diff
@@ -123,6 +153,25 @@ def _init(self, config=None, merge=False, **kwargs):
                 RemoteOutput(self.cmd_manager, '<stderr>'),
             )
 
+    def bound_cmd_manager():
+        def fget(self):
+            return self.__cmd_manager
+
+        class BoundCmdMangaer(object):
```

Comment:
If you don't know the pattern that's being used here, it's really hard to grok what bound_cmd_manager is.  Particularly confusing is the fact that you're defining a function and then overwriting that function with itself in the class namespace.  I'd rename the function here to something like `__make_bound_cmd_manager_property`, and then del it out of the namespace if you're really worried about the overhead (hint: you probably shouldn't be).